### ISSUE TITLE
Stubbing real API requests in test env

### ISSUE THREAD

**Issue by FunkyloverOne** [spk/money-open-exchange-rates#53]
Hey, I don't see that there is any guide on how to use this bank in app tests without sending real requests to OXR API.
I guess that the best option for now is to stub `OpenExchangeRatesBank#api_response` method.
What do you think about that @spk ?

It would be good to have some instructions for it in README, or even better - have a support for testing out of the box, in a similar way to how sidekiq does it, here's more info:
https://github.com/mperham/sidekiq/wiki/Testing

**Comment by spk** [comment]
Hi @FunkyloverOne yes good idea I usually download and use use the `oxr.cache = 'path/to/file/cache.json'` option on specs to avoid to query the OXR API, I will try to write about it soon

**Comment by FunkyloverOne** [comment]
OK, meanwhile I have this setup (I use `master` version as it has "symbols" support):
```ruby
  oxr = Money::Bank::OpenExchangeRatesBank.new
  oxr.app_id = ENV['open_exchange_rates_app_id']

  oxr.symbols = %w[RUB]

  # 24 hours
  oxr.ttl_in_seconds = 86_400

  oxr.cache =
    if Rails.env.test?
      Rails.root.join(
        'spec', 'stubs', 'open_exchange_rates_11-jan-2019.json'
      ).to_s
    else
      proc do |text|
        if text
          # Make it non-expirable
          Rails.cache.write(OXR_CACHE_KEY, text, expires_in: -1)
        else
          Rails.cache.read(OXR_CACHE_KEY)
        end
      end
    end

  oxr.update_rates

  config.default_bank = oxr
```

BTW, I've noticed that it is also possible to get minified JSON response, so it would be great to add that to upcoming release as well :)
Here's more info: https://docs.openexchangerates.org/docs/prettyprint

**Comment by FunkyloverOne** [comment]
Also there is a small gotcha:
I tried to edit my stub JSON to remove all the currencies except RUB, but left trailing comma inside `"rates"`, and my stub JSON was considered not valid, and the requests to OXR API were still sending, so I were reading the source code of this gem to figure out why is that happening :)

So maybe just in case it's worth mentioning not to do this mistake:
```javascript
{
  "disclaimer": "Usage subject to terms: https://openexchangerates.org/terms",
  "license": "https://openexchangerates.org/license",
  "timestamp": 1547229596,
  "base": "USD",
  "rates": {
    "RUB": 66.8696, // <--- do not leave this comma here!
  }
}
```

**Comment by spk** [comment]
Hi @FunkyloverOne I've added some info on the README for the tests in https://github.com/spk/money-open-exchange-rates/commit/82521df35f9fb29fe74146c45f28489016147258 also did the prettyprint option you reported here, will do a release soon with this

**Comment by FunkyloverOne** [comment]
OK, I guess that's just enough, so this issue may be closed now.

**Comment by spk** [comment]
Released in [v1.3.0](https://github.com/spk/money-open-exchange-rates/releases/tag/v1.3.0)
